papers/whitepaper.md

```diff
@@ -111,3 +113,141 @@ The focus of the current text is to explore at a deeper level the Atoma's unique
+
+On the other hand, if a node acts maliciously, it gets penalized with all of its deposited collateral being slashed. Similarly to the timeout case above, a percentage of the slashed collateral goes to the user, compensating for any possible malicious behavior. There is also a percentage of the slashed collateral that is allocated to the honest nodes involved in processing the request, the Atoma nodes that participated in the dispute event, to the Atoma community treasury fund, and the remaining collateral is burned.
+
+On the other hand, for each processed request, a node is entitled to part of the fee paid by the user that initiated the request. Request fees are paid in `TOMA` tokens and their price is set by a fair free market mechanism. At the end of each epoch, the Atoma nodes bid the TOMA token fee value, per token, for the next epoch. Therefore, fees are computed in terms of the total number of tokens a given request requires to be processed. So for example, if a prompt text request has 100 input tokens, and its output response contains 200 tokens, the total fee amount paid to a node to process that request will be none other than:
+
+$(100 + 200) * \mathrm{token\_fee} = 300 * \mathrm{token\_fee}$.
+
+Nodes accrue fees per epoch. Moreover, nodes can only withdraw their accrued fees for a given epoch, once that epoch has ended.
+
+**Node selection:** For every request submitted into the Atoma Network, the Atoma on-chain logic is responsible for selecting one or more nodes to process that request. A node is selected uniformly at random, from the set of all the nodes that are subscribed to the AI model that the request requires. This random sampling mechanism helps balancing request load and ensures a fair
+distribution of the fees being accrued by nodes. 
+
+**Sampling Consensus:** Moreover, a request might require multiple nodes to process it, in order to provide high assurance guarantees that the generated response was not tampered with. In this case, the request is only settled by the Atoma on-chain logic, once all the nodes submitted a commitment to the output state, and these commitments all agree. More precisely, for a given request and a selected node to process it, that node must compute the output generated by running the AI model inference on the request's input. Once the output `O` is generated, the node must then submit a short cryptographic commitment to the output `O`, in the form of:
```

Comment:
A request might require multiple nodes to process it in order to provide high assurance guarantees that the generated response was not tampered with. In this case, the request is only settled by the Atoma on-chain logic once all the nodes have submitted a commitment to the output state and these commitments all agree.